src/bridge/cockpitpipechannel.c

```diff
@@ -415,11 +415,17 @@ cockpit_pipe_channel_prepare (CockpitChannel *channel)
   const gchar *internal = NULL;
   const gchar *dir;
   const gchar *error;
+  gchar *debug_msg;
+  gsize length;
+
   gint fd;
 
   COCKPIT_CHANNEL_CLASS (cockpit_pipe_channel_parent_class)->prepare (channel);
 
   options = cockpit_channel_get_options (channel);
+￼
+  debug_msg = cockpit_json_write_object (options, &length);
+  g_warning ("Debug Options: %s", debug_msg);
```

Comment:
A warning will break every unit test that touches this, so maybe reduce it to a `g_message()`. But actually, I think it makes more sense to move this into the `cockpit_channel_fail()` call, like so:
```
        cockpit_channel_fail (channel, "protocol-error",
                            "invalid \"spawn\" option for stream channel: %s", cockpit_json_write_object (options, NULL));
```